Services/APIVersioning.md

```diff
@@ -0,0 +1,144 @@
+
+## Clients should conform to the HTTP Spec
+
+While the whole spec is important, it is worth calling out specific common cases where we expect clients to conform: 
+
+* Clients should follow HTTP-level redirects when a resource has moved to a new location.
+* Clients should obey any caching directives returned with a resource
+* If a client actively engages in content or language negotiation, it should account for the possibility that a new representation of the resource will become available in the future that is a better match.
+
+## Clients should use appropriate parsers for the content they receive.
+
+[Do not parse XHTML with regular expressions](http://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454).
+
+## Clients should ignore parts of a response or representation they do not explicitly expect
+
+When a REST resource returns structured information, new elements may be added to that structure at any time. Clients should expect and ignore this, and constrain themselves to looking for the information they expect.
+
+## Clients should handle unexpected values in data gracefully, so long as the structure and type of that data remains the same.
+
+For example, say a REST resource returns:
+
+```json
+    {"color": "red"}
+```
+
+Even if the only (currently) documented colors are "red", "green" and "blue", the client should anticipate the possibility of the value being _any_ legal JSON string, including "purple", "taupe" or "My Cousin Vinny".
+
+## External clients will more often than not be somebody's quick hack
+
+While we necessarily make the above assumptions about third-party clients, we recognise that they're not going to be perfect:
+
+* They will generate URIs using string concatenation even when the URI structure is not documented
+* They will not be updated while they still work
+* If they break, they are just as likely to be abandoned as updated
+
+# Responsibilities of Wikia-Authored Clients
+
+While we can not control the quality of third-party clients, we _can_ control the quality of our own code. By holding ourselves to higher standards we give ourselves more flexibility to evolve APIs and services that are only used internally.
+
+## All "should" requirements for third-party clients are "MUST" requirements for internal clients.
+
+While third-party developers shouldn’t write code that breaks when a conforming service is upgraded, we mustn't.
+
+## Clients should not make assumptions about an API's URI structure
+
+Clients should not engage in "URI guessing" by using manual URI generation to reach resources unless doing so is explicitly documented as how the API is intended to be used. Whereever possible, clients should discover resource by following links from within the API itself (see the section below on HATEOAS).
```

Comment:
"…unless doing so is explicitly documented as how the API is intended to be used".

Nothing in this document absolves developers from the responsibility of thinking about the usage patterns of their service and designing an APIs that satisfies those uses.

A good example of this is Vignette - because most hits to the service are "one-off" accesses of generated resources via GET, and because we have to support its URLs indefinitely anyway to preserve deep links, we chose to make those URLs part of the API.

As for the rest, I'm going to remove the section on HATEOAS and work on a separate explanation of it as while I do have clarifications of your various points, it seems I would have to clutter this document up significantly in order to include them.